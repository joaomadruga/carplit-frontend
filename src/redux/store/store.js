import { createContext } from "react";

export const CarpoolContext = createContext();
export const LoginContext = createContext();
export const HomeContext = createContext();