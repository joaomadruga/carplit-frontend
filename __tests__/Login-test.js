import { fireEvent, render, screen } from '@testing-library/react-native';
import React from 'react';
import { create, act } from 'react-test-renderer';
import LoginScreen from '../src/views/login';
import * as Store from '../src/redux/store/store';

const isLogin = false;
const loginInfo = "pedro@gmail.com";
const password = "12345678";

test('renders login correctly', () => {
    const telaentrada = create(
    <Store.LoginContext.Provider value={{ loginInfo, password, isLogin }}>  
        <LoginScreen />
    </Store.LoginContext.Provider>).toJSON();
    expect(telaentrada).toMatchSnapshot();
    });

it('login entrance button title', () => {
    const title = 'Entrar';
    const loginbutton = render(
    <Store.LoginContext.Provider value={{ loginInfo, password, isLogin }}>  
        <LoginScreen />
    </Store.LoginContext.Provider>).toJSON();
    const foundButtonTitle = screen.getByText(title);
    expect(foundButtonTitle.props.children).toEqual(title);
});
      
it('forgot password text', () => {
    const title = 'Esqueci minha senha';
    const forgetpasstext = render(
    <Store.LoginContext.Provider value={{ loginInfo, password, isLogin }}>  
        <LoginScreen />
    </Store.LoginContext.Provider>).toJSON();
    const foundButtonTitle = screen.getByText(title);
    expect(foundButtonTitle.props.children).toEqual(title);
});

it('render default elements', () => {
    const entrance = render(
    <Store.LoginContext.Provider value={{ loginInfo, password, isLogin }}>  
        <LoginScreen />
    </Store.LoginContext.Provider>).toJSON();
    const qtd = screen.getAllByText('Entrar').length;
    expect(qtd).toBe(1);
});

it('render default elements', () => {
    const entrance = render(
    <Store.LoginContext.Provider value={{ loginInfo, password, isLogin }}>  
        <LoginScreen />
    </Store.LoginContext.Provider>).toJSON();
    const qtd = screen.getAllByText('Esqueci minha senha').length;
    expect(qtd).toBe(1);
});

it('login testing placeholder', () => {
    const component = render(
    <Store.LoginContext.Provider value={{ loginInfo, password, isLogin }}>  
        <LoginScreen />
    </Store.LoginContext.Provider>).toJSON();
    const input = screen.getByPlaceholderText('Login');
    expect(input.props.placeholder).toBe('Login');
  });

it('password testing placeholder', () => {
    const component = render(
    <Store.LoginContext.Provider value={{ loginInfo, password, isLogin }}>  
        <LoginScreen />
    </Store.LoginContext.Provider>).toJSON();
    const input = screen.getByPlaceholderText('Senha');
    expect(input.props.placeholder).toBe('Senha');
  });

it('error message', () => {
    const buttonLogin = render(
    <Store.LoginContext.Provider value={{ loginInfo, password, isLogin }}>  
        <LoginScreen />
    </Store.LoginContext.Provider>).toJSON();
    fireEvent.press(screen.getByTestId("test"))
});
