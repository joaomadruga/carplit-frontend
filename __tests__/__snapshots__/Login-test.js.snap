// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders login correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "flexGrow": 1,
        "height": "100%",
        "width": "100%",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "display": "flex",
          "height": "100%",
          "paddingBottom": 0,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 0,
          "width": "100%",
        },
        Object {
          "alignItems": "center",
          "marginTop": 40,
        },
      ]
    }
  >
    <TextInput
      marginBottom={12}
      onChangeText={[Function]}
      placeholder="Login"
      placeholderTextColor="#686D73"
      style={
        Array [
          Object {
            "backgroundColor": "#F2F2F3",
            "borderRadius": 8,
            "color": "",
            "fontFamily": "RetniSans-Medium",
            "fontSize": 16,
            "height": 64,
            "marginBottom": 12,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 0,
            "maxWidth": 500,
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
            "width": "100%",
          },
        ]
      }
    />
    <TextInput
      marginBottom={38}
      onChangeText={[Function]}
      placeholder="Senha"
      placeholderTextColor="#686D73"
      secureTextEntry={true}
      style={
        Array [
          Object {
            "backgroundColor": "#F2F2F3",
            "borderRadius": 8,
            "color": "",
            "fontFamily": "RetniSans-Medium",
            "fontSize": 16,
            "height": 64,
            "marginBottom": 38,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 0,
            "maxWidth": 500,
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
            "width": "100%",
          },
        ]
      }
    />
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#6E1DBE",
            "borderRadius": 8,
            "display": "flex",
            "height": 64,
            "justifyContent": "center",
            "marginBottom": 0,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 0,
            "maxWidth": 500,
            "width": "100%",
          },
          Object {
            "backgroundColor": "#6E1DBE",
          },
        ]
      }
      testID="test"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "RetniSans-Bold",
              "fontSize": 16,
            },
          ]
        }
      >
        Entrar
      </Text>
    </View>
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#6E1DBE",
            "fontFamily": "RetniSans-Bold",
            "fontSize": 16,
            "marginTop": 30,
            "textAlign": "center",
            "width": "100%",
          },
        ]
      }
      suppressHighlighting={true}
    >
      Esqueci minha senha
    </Text>
  </View>
</RCTSafeAreaView>
`;
